clients/src/test/java/org/apache/kafka/common/record/SimpleRecordTest.java

```diff
@@ -35,8 +37,9 @@
     public void testCompressedIterationWithNullValue() throws Exception {
         ByteBuffer buffer = ByteBuffer.allocate(128);
         DataOutputStream out = new DataOutputStream(new ByteBufferOutputStream(buffer));
-        LogEntry.writeHeader(out, 0L, Record.RECORD_OVERHEAD_V1);
-        Record.write(out, Record.CURRENT_MAGIC_VALUE, 1L, null, null, CompressionType.GZIP, TimestampType.CREATE_TIME);
+        AbstractLegacyRecordBatch.writeHeader(out, 0L, LegacyRecord.RECORD_OVERHEAD_V1);
```

Comment:
I think we should rename this class `LegacySimpleRecordTest` (or something like that) and move the tests that are not for `LegacyRecord` elsewhere. Also, if there are tests that would make sense for `DefaultRecord`, we should port them.